app/processing/statement/components/get-detailed-payments.js

```diff
@@ -0,0 +1,18 @@
+
+const getDetailedPayments = async (calculation, paymentRequest, settlement) => {
+  const payments = []
+
+  const payment = {
+    invoiceNumber: calculation.invoiceNumber,
+    reference: settlement.reference,
+    dueDate: paymentRequest.dueDate,
+    settled: settlement.settled,

```

Comment:
These date values need to be formatted from the below formats

```
"invoiceNumber":"S0000001C000001V001",
         "reference":"PY1234567",
         "dueDate":"1 July 2022",
         "settled":"1 July 2022",
         "calculated":"16 June 2022",
         "value":"317.15"
```

Different approach for each as due date is a string, but settled is a date.

Moment to the rescue?

Reply:
Points to this function having unit tests around this which suggests moving it into its own file

Reply:
Moved to new ticket https://eaflood.atlassian.net/browse/PAY-2364